src/script/interpreter.cpp

```diff
@@ -1279,18 +1279,28 @@ uint256 GetOutputsHash(const T& txTo)
 } // namespace
 
 template <class T>
-PrecomputedTransactionData::PrecomputedTransactionData(const T& txTo)
+void PrecomputedTransactionData::Init(const T& txTo)
 {
     // Cache is calculated only for transactions with witness
-    if (txTo.HasWitness()) {
+    if (!txTo.HasWitness()) return;
+
+    if (!ready) {
```

Comment:
while we're touching this code can we make `ready` be something more self-explanatory? Especially if we're expanding condition checks in the future.

Reply:
This, plus the discussion here: https://github.com/bitcoin/bitcoin/pull/18401/files#r401812158 has made me think that there shouldn't be two different ready bools in the final taproot implementation, since we only ever want to initialize the object once. I've changed this to `m_ready` and made some minor changes to the `Init` function.